lntest/itest/lnd_taproot_test.go

```diff
@@ -80,4 +89,153 @@ func testTaproot(net *lntest.NetworkHarness, t *harnessTest) {
+	keyDesc, err := net.Alice.WalletKitClient.DeriveNextKey(
+		ctxt, &walletrpc.KeyReq{
+			KeyFamily: taprootKeyFamily,
+		},
+	)
+	require.NoError(t.t, err)
+
+	leafSigningKey, err := btcec.ParsePubKey(keyDesc.RawKeyBytes)
+	require.NoError(t.t, err)
+
+	// Let's create a taproot script output now. This is a hash lock with a
+	// simple preimage of "foobar".
+	builder := txscript.NewScriptBuilder()
+	builder.AddOp(txscript.OP_DUP)
+	builder.AddOp(txscript.OP_HASH160)
+	builder.AddData(btcutil.Hash160([]byte("foobar")))
+	builder.AddOp(txscript.OP_EQUALVERIFY)
+	script1, err := builder.Script()
+	require.NoError(t.t, err)
+	leaf1 := txscript.NewBaseTapLeaf(script1)
+
+	// Let's add a second script output as well to test the partial reveal.
+	builder = txscript.NewScriptBuilder()
+	builder.AddData(schnorr.SerializePubKey(leafSigningKey))
+	builder.AddOp(txscript.OP_CHECKSIG)
+	script2, err := builder.Script()
+	require.NoError(t.t, err)
+	leaf2 := txscript.NewBaseTapLeaf(script2)
+
+	dummyInternalKeyBytes, _ := hex.DecodeString(
+		"03464805f5468e294d88cf15a3f06aef6c89d63ef1bd7b42db2e0c74c1aceb90fe",
+	)
+	dummyInternalKey, _ := btcec.ParsePubKey(dummyInternalKeyBytes)
+
+	tree := txscript.AssembleTaprootScriptTree(leaf1, leaf2)
+	rootHash := tree.RootNode.TapHash()
+	taprootKey := txscript.ComputeTaprootOutputKey(
+		dummyInternalKey, rootHash[:],
+	)
+
+	tapScriptAddr, err := btcutil.NewAddressTaproot(
+		schnorr.SerializePubKey(taprootKey), harnessNetParams,
+	)
+	require.NoError(t.t, err)
+	p2trPkScript, err := txscript.PayToAddrScript(tapScriptAddr)
+	require.NoError(t.t, err)
+
+	// Send some coins to the generated tapscript address.
+	_, err = net.Alice.SendCoins(ctxt, &lnrpc.SendCoinsRequest{
+		Addr:   tapScriptAddr.String(),
+		Amount: 800_000,
+	})
+	require.NoError(t.t, err)
+
+	// Wait until the TX is found in the mempool.
+	txid, err = waitForTxInMempool(net.Miner.Client, minerMempoolTimeout)
+	require.NoError(t.t, err)
+
+	p2trOutputIndex = getOutputIndex(t, net, txid, tapScriptAddr.String())
+
+	// Spend the output again, this time back to a p2wkh address.
+	p2wkhResp, err := net.Alice.NewAddress(ctxt, &lnrpc.NewAddressRequest{
+		Type: lnrpc.AddressType_WITNESS_PUBKEY_HASH,
+	})
+	require.NoError(t.t, err)
+
+	p2wkhAdrr, err := btcutil.DecodeAddress(
+		p2wkhResp.Address, harnessNetParams,
+	)
+	require.NoError(t.t, err)
+
+	p2wkhPkScript, err := txscript.PayToAddrScript(p2wkhAdrr)
+	require.NoError(t.t, err)
+
+	tx := wire.NewMsgTx(2)
+	tx.TxIn = []*wire.TxIn{{
+		PreviousOutPoint: wire.OutPoint{
+			Hash:  *txid,
+			Index: uint32(p2trOutputIndex),
+		},
+	}}
+	value := int64(800_000 - 500) // TODO(guggero): Calculate actual fee.
+	tx.TxOut = []*wire.TxOut{{
+		PkScript: p2wkhPkScript,
+		Value:    value,
+	}}
+
+	var buf bytes.Buffer
+	require.NoError(t.t, tx.Serialize(&buf))
+
+	utxoInfo := []*signrpc.TxOut{{
+		PkScript: p2trPkScript,
+		Value:    800_000,
+	}}
+	signResp, err := net.Alice.SignerClient.SignOutputRaw(
```

Comment:
I don't see where in this diff we allow the user to specify a _schnorr_ signature here?